enclave-runtime/src/initialization/parentchain/parachain.rs

```diff
@@ -0,0 +1,115 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::{
+	error::Result,
+	initialization::{
+		global_components::{
+			EnclaveExtrinsicsFactory, EnclaveNodeMetadataRepository, EnclaveOCallApi,
+			EnclaveParentchainBlockImportDispatcher, EnclaveStfExecutor, EnclaveValidatorAccessor,
+			GLOBAL_FULL_PARACHAIN_HANDLER_COMPONENT, GLOBAL_OCALL_API_COMPONENT,
+			GLOBAL_STATE_HANDLER_COMPONENT,
+		},
+		parentchain::common::{
+			create_extrinsics_factory, create_offchain_immediate_import_dispatcher,
+			create_parentchain_block_importer, create_sidechain_triggered_import_dispatcher,
+			HeaderFor,
+		},
+	},
+};
+use codec::{Decode, Encode};
+use itc_parentchain::light_client::{
+	concurrent_access::ValidatorAccess, light_client_init_params::ParachainParams, LightClientState,
+};
+use itp_component_container::{ComponentGetter, ComponentInitializer};
+use itp_settings::worker_mode::{ProvideWorkerMode, WorkerMode};
+use itp_types::Block as ParentchainBlock;
+use std::{sync::Arc, vec::Vec};
+
+#[derive(Clone)]
+pub struct FullParachainHandler {
+	pub genesis_header: HeaderFor<ParentchainBlock>,
+	pub node_metadata_repository: Arc<EnclaveNodeMetadataRepository>,
+	pub stf_executor: Arc<EnclaveStfExecutor>,
```

Comment:
I have not checked the low-level, but why does this need the `stf_executor`?

Reply:
The stf_executor currently executes parentchain (indirect) extrinsics, so it is a parentchain dependent component.
I placed a FIXME comment at the initilzation part:

`// FIXME: Probably should be split up into a parentchain dependent executor and one independent.`

Reply:
Moved the comment to be better visible. Is that okay?

Reply:
Thanks for adding the comment. I do think that this should be separated, the `ParentchainHandler` should only forward the indirect request. This gives us a cleaner separation of concerns.